libsolutil/IpfsHash.cpp

```diff
@@ -53,36 +53,133 @@ string base58Encode(bytes const& _data)
 	reverse(output.begin(), output.end());
 	return output;
 }
+struct Chunk
+{
+	Chunk() = default;
+	Chunk(bytes _hash, size_t _size, size_t _blockSize):
+		hash(std::move(_hash)),
+		size(_size),
+		blockSize(_blockSize),
+		links{},
+		lengths{}
+	{}
+	bytes hash = {};
+	size_t size = 0;
+	size_t blockSize = 0;
+
+	bytes links = {};
```

Comment:
Is it possible to separate blocks and links? Also the `updateHash` does not sound like a good idea.

Reply:
When you say blocks and links, you mean data chunks and link chunks?

Reply:
I think so